tests/jax/components/executing/observing_test.py

```diff
@@ -0,0 +1,272 @@
+            [agent_net_keys[key] for key in sort_str_num(agent_net_keys.keys())]
+        ]
+        # net_keys_to_ids
+        all_samples = []
+        for sample in network_sampling_setup:
+            all_samples.extend(sample)
+        unique_net_keys = list(sort_str_num(list(set(all_samples))))
+        net_keys_to_ids = {net_key: i for i, net_key in enumerate(unique_net_keys)}
+        # network_int_keys_extras
+        network_int_keys_extras = None
+        # timestep
+        timestep = TimeStep(
+            step_type=StepType.FIRST,
+            reward=0.0,
+            discount=1.0,
+            observation=OLT(
+                observation=[0.1, 0.3, 0.7], legal_actions=[1], terminal=[0.0]
+            ),
+        )
+        # extras
+        extras = {}
+        # Aadder
+        adder = MockAdder()
+        # actions_info
+        actions_info = {
+            "agent_0": "action_info_agent_0",
+            "agent_1": "action_info_agent_1",
+            "agent_2": "action_info_agent_2",
+        }
+        # policies_info
+        policies_info = {
+            "agent_0": "policy_info_agent_0",
+            "agent_1": "policy_info_agent_1",
+            "agent_2": "policy_info_agent_2",
+        }
+        # executor_parameter_client
+        executor_parameter_client = MockExecutorParameterClient()
+        # Store
+        store = SimpleNamespace(
+            is_evaluator=None,
+            observations={},
+            policy={},
+            agent_net_keys=agent_net_keys,
+            network_int_keys_extras=network_int_keys_extras,
+            network_sampling_setup=network_sampling_setup,
+            net_keys_to_ids=net_keys_to_ids,
+            timestep=timestep,
+            extras=extras,
+            adder=adder,
+            next_extras=extras,
+            next_timestep=timestep,
+            actions_info=actions_info,
+            policies_info=policies_info,
+            executor_parameter_client=executor_parameter_client,
+        )
+        self.store = store
+
+
+@pytest.fixture
+def mock_executor() -> MockExecutor:
+    """Mock executor component."""
+    return MockExecutor()
+
+
+@pytest.fixture
+def mock_feedforward_executor_observe() -> FeedforwardExecutorObserve:
+    """Mock FeedforwardExecutorObserve.
+
+    Returns:
+        FeedforwardExecutorObserve
+    """
+    return FeedforwardExecutorObserve()
+
+
+# Test on_execution_observe_first
+def test_on_execution_observe_first_without_adder(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor_without_adder: Executor,
+) -> None:
+    """Test entering executor without store.adder
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor_without_adder: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe_first(
+        executor=mock_executor_without_adder
+    )
+
+    assert not mock_executor_without_adder.store.adder
```

Comment:
assert not hasattr(mock_executor_without_adder.store,"extras") I think this will solve the problem, what do you think?

Reply:
It won't solve it, I think, because `extras` is already set in the store by the executor. What is set by the method you're trying to test is
```
executor.store.extras[
            "network_int_keys"
        ] = executor.store.network_int_keys_extras
```
You need to ensure that this code is not run if you do not provide an adder.

Reply:
mock_executor_without_adder is a function that return a real exacutor not a mock Executor(store=store) so shouldn't just verify that the store has no betwork_int_keys_extra is enough?

Reply:
I think an easy way to test this is to assert that `executor.store` is the same before and after the hook is called.